### PR TITLE
Drop support for /dev/.udev and like

### PR DESCRIPTION
**Pull request by bigon**
This location is gone for quite some times and the udevdb has been moved
to /run/udev.

Drop the udev_tbl_t and deprecate the udev_read_db() function

This inspired from changes in the Red Hat policy

Signed-off-by: Laurent Bigonville <bigon@bigon.be>

Fixes: #221

### CODE REVIEW

File: policy/modules/apps/pulseaudio.te [hunk 1]
```diff
@@ -235,7 +235,7 @@ optional_policy(`
 optional_policy(`
 	udev_read_pid_files(pulseaudio_t)
 	udev_read_state(pulseaudio_t)
-	udev_read_db(pulseaudio_t)
+	udev_read_pid_files(pulseaudio_t)
```
Comment (pebenito) [inline policy/modules/apps/pulseaudio.te:238]: Already has the permission.

This prompts a broader question: If the domain doesn't already have the /run access, maybe there no longer is a need? Perhaps we should be dropping the permission instead.

File: policy/modules/services/colord.te [hunk 2]
```diff
@@ -133,7 +133,6 @@ optional_policy(`
 ')
 
 optional_policy(`
-	udev_read_db(colord_t)
```
Comment (pebenito) [inline policy/modules/services/colord.te:136]: Why was this removed, but the others switched over to the pid one?

Same question for consolekit below.

File: policy/modules/services/modemmanager.te [hunk 3]
```diff
@@ -55,6 +55,6 @@ optional_policy(`
 ')
 
 optional_policy(`
-	udev_read_db(modemmanager_t)
+	udev_read_pid_files(modemmanager_t)
 	udev_manage_pid_files(modemmanager_t)
```
Comment (pebenito) [inline policy/modules/services/modemmanager.te:59]: Already has more permissions.

File: policy/modules/system/logging.te [hunk 4]
```diff
@@ -607,7 +607,7 @@ optional_policy(`
 ')
 
 optional_policy(`
-	udev_read_db(syslogd_t)
+	udev_read_pid_files(syslogd_t)
 	# for systemd-journal to read seat data from /run/udev/data
 	udev_read_pid_files(syslogd_t)
```
Comment (pebenito) [inline policy/modules/system/logging.te:612]: Already has the perms

File: policy/modules/services/colord.te (hunk 2 above)
Comment (bigon) [reply policy/modules/services/colord.te:136]: The call to `udev_read_pid_files(colord_t)` is just bellow :)

File: policy/modules/system/udev.if [hunk 5]
```diff
@@ -214,10 +214,10 @@ interface(`udev_manage_rules_files',`
 #
 interface(`udev_dontaudit_search_db',`
 	gen_require(`
-		type udev_tbl_t;
+		type udev_runtime_t;
 	')
 
-	dontaudit $1 udev_tbl_t:dir search_dir_perms;
+	dontaudit $1 udev_runtime_t:dir search_dir_perms;
```
Comment (pebenito) [inline policy/modules/system/udev.if:220]: This interface should be deprecated too.

**Comment by pebenito** [comment]
Is there any progress expected on this?

**Comment by bigon** [comment]
I think that the rework should still be done and drop support for the ancient `/dev/.udev` location but I'm not sure I'll have time to continue on this

**Comment by bigon** [comment]
Thx
